extensions/pkg/controller/heartbeat/reconciler.go

```diff
@@ -47,27 +47,20 @@ func (r *reconciler) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 			Name:      extensions.HeartBeatResourceName,
 			Namespace: r.namespace,
 		},
+		Spec: coordinationv1.LeaseSpec{
+			HolderIdentity:       &r.extensionName,
+			LeaseDurationSeconds: &r.renewIntervalSeconds,
+			RenewTime:            &metav1.MicroTime{Time: r.clock.Now().UTC()},
+		},
 	}
 
 	if err := r.client.Get(ctx, client.ObjectKeyFromObject(lease), lease); err != nil {
 		if apierrors.IsNotFound(err) {
-			lease.Spec = coordinationv1.LeaseSpec{
-				HolderIdentity:       &r.extensionName,
-				LeaseDurationSeconds: &r.renewIntervalSeconds,
-				RenewTime:            &metav1.MicroTime{Time: r.clock.Now().UTC()},
-			}
```

Comment:
Isn't the `lease` object "reset" when the object is not found? Maybe we should rather use `controllerutil.CreateOrUpdate`?

Reply:
Additionally, when the object exists, the renew time will be overwritten with the one retrieved from the server.

With `controllerutil.CreateOrUpdate` we loose the differing log messages, but I don't have too strong of an opinion on that :D.

Reply:
I personally don't think the difference of the log messages provides any actionable difference